app/data/domain/exemption.js

```diff
@@ -19,33 +22,62 @@ const { ChangeManager } = require('./changeManager')
-  this.cdoExpiry = exemptionProperties.cdoExpiry
-  this.court = exemptionProperties.court
-  this.policeForce = exemptionProperties.policeForce
-  this.legislationOfficer = exemptionProperties.legislationOfficer
-  this.certificateIssued = exemptionProperties.certificateIssued
-  this.applicationFeePaid = exemptionProperties.applicationFeePaid
-  this.insurance = exemptionProperties.insurance
-  this.neuteringConfirmation = exemptionProperties.neuteringConfirmation
-  this.microchipVerification = exemptionProperties.microchipVerification
-  this.joinedExemptionScheme = exemptionProperties.joinedExemptionScheme
-  this.nonComplianceLetterSent = exemptionProperties.nonComplianceLetterSent
-  this.applicationPackSent = exemptionProperties.applicationPackSent
-  this.formTwoSent = exemptionProperties.formTwoSent
-}
+class Exemption {
+  constructor (exemptionProperties) {
+    this._updates = new ChangeManager()
+    this.exemptionOrder = exemptionProperties.exemptionOrder
+    this.cdoIssued = exemptionProperties.cdoIssued
+    this.cdoExpiry = exemptionProperties.cdoExpiry
+    this.court = exemptionProperties.court
+    this.policeForce = exemptionProperties.policeForce
+    this.legislationOfficer = exemptionProperties.legislationOfficer
+    this.certificateIssued = exemptionProperties.certificateIssued
+    this.applicationFeePaid = exemptionProperties.applicationFeePaid
+    this._insurance = exemptionProperties.insurance
+    this.neuteringConfirmation = exemptionProperties.neuteringConfirmation
+    this.microchipVerification = exemptionProperties.microchipVerification
+    this.joinedExemptionScheme = exemptionProperties.joinedExemptionScheme
+    this.nonComplianceLetterSent = exemptionProperties.nonComplianceLetterSent
+    this.applicationPackSent = exemptionProperties.applicationPackSent
+    this.formTwoSent = exemptionProperties.formTwoSent
+  }
 
-Exemption.prototype.sendApplicationPack = function (callback) {
-  const auditDate = new Date()
-  this.applicationPackSent = auditDate
-  this._updates.update('applicationPackSent', auditDate, callback)
-}
+  sendApplicationPack (auditDate, callback) {
+    this.applicationPackSent = auditDate
+    this._updates.update('applicationPackSent', auditDate, callback)
+  }
 
-Exemption.prototype.getChanges = function () {
-  return this._updates.changes
-}
+  getChanges () {
+    return this._updates.changes
+  }
+
+  get insurance () {
+    return this._insurance
+  }
+
+  setInsuranceDetails (company, insuranceRenewal, callback) {
+    if (insuranceRenewal instanceof Date && !dateTodayOrInFuture(insuranceRenewal)) {
+      throw new InvalidDateError('Insurance renewal date must be in the future')
+    }
 
+    const insurance = {
+      company: company || undefined,
+      insuranceRenewal
+    }
+
+    if (insurance.company === undefined && insuranceRenewal instanceof Date) {
+      throw new IncompleteDataError('Insurance company must be submitted')
```

Comment:
How are we handling these errors? i.e. is the UI displaying their text content? Does the content need agreeing with Nick/Matt?

Reply:
I don't think they are handled yet on the UI side.  Provided the UI is solid, which I believe it is, this shouldn't be reachable.